webui/react/src/hooks/useResize.ts

```diff
@@ -0,0 +1,43 @@
+import { RefObject, useEffect, useState } from 'react';
+
+interface ResizeInfo {
+  height: number;
+  width: number;
+  x: number;
+  y: number;
+}
+
+const defaultResizeInfo = {
+  height: 0,
+  width: 0,
+  x: 0,
+  y: 0,
+};
+
+export const DEFAULT_RESIZE_THROTTLE_TIME = 500;
+
+const useResize = (ref: RefObject<HTMLElement>): ResizeInfo => {
+  const element = ref.current;
+  const [ resizeInfo, setResizeInfo ] = useState<ResizeInfo>(defaultResizeInfo);
+
+  useEffect(() => {
+    if (!element) return;
+
+    const handleResize: ResizeObserverCallback = entries => {
+      // Check to make sure the log viewer container is being observed for resize.
```

Comment:
question: is the component-specific to log viewer or you think we'd use it elsewhere?

Reply:
great question. It's currently being used in two places: `LogViewer` and `MetricChart`, but I can see future components that need to detect changes in order to redraw the component via JS (non CSS layouts).

Reply:
I'll update the comment :)